flytekit/annotated/workflow.py

```diff
@@ -358,7 +389,11 @@ def get_registerable_entity(self) -> _SdkWorkflow:
         return self._registerable_entity
 
 
-def workflow(_workflow_function=None, *args, reference: Optional[WorkflowReference] = None, **kwargs):
+def workflow(
+    _workflow_function=None,
+    reference: Optional[WorkflowReference] = None,
```

Comment:
the workflow has both - reference and metadata. I guess we will keep it like task. That was one though I had, should we just expose the `WorkflowMetadata` `TaskMetadata` objects to the user?